web/modules/custom/dpl_library_agency/src/Branch/FbsBranchRepository.php

```diff
@@ -13,22 +14,29 @@ class FbsBranchRepository implements BranchRepositoryInterface {
   /**
    * External branch API constructor.
    *
-   * @param \DanskernesDigitaleBibliotek\FBS\Api\ExternalV1AgencyidApi $api
-   *   The API instance to use to integrate with FBS.
+   * @param \Drupal\dpl_fbs\FbsApiFactory $apiFactory
+   *   API factory to setup the agency API.
+   * @param \Drupal\dpl_library_token\LibraryTokenHandler $tokenHandler
+   *   The token handler to use to retrieve the library token.
    */
   public function __construct(
-    private ExternalV1AgencyidApi $api,
-  ) {}
+    private FbsApiFactory $apiFactory,
+    private LibraryTokenHandler $tokenHandler,
+  ) {
+  }
 
   /**
    * {@inheritdoc}
    *
    * @throws \DanskernesDigitaleBibliotek\FBS\ApiException
    */
   public function getBranches(): array {
+    // Use an empty token if we don't have one. This may cause the API to throw
+    // an exception, but that is acceptable.
+    $api = $this->apiFactory->getAgencyApi($this->tokenHandler->getToken() ?? '');
```

Comment:
I think it would be prettier if the AgencyApi could handle that there should be a token for its requests instead of passing it as an argument.